app/controllers/page_views_controller.rb

```diff
@@ -18,8 +18,10 @@ def create
 
   def update
     if session_current_user_id
-      page_view = PageView.find_or_create_by(article_id: params[:id], user_id: session_current_user_id)
-      page_view.update_column(:time_tracked_in_seconds, page_view.time_tracked_in_seconds + 15)
+      page_view = PageView.order("created_at DESC").find_or_create_by(article_id: params[:id], user_id: session_current_user_id)
+      unless page_view.new_record?
+        page_view.update_column(:time_tracked_in_seconds, page_view.time_tracked_in_seconds + 15)
+      end
```

Comment:
what if we use the block version so that it his the DB only once?

```ruby
PageView.find_or_create_by(params) do |page_view|
  page_view.time_tracked_in_seconds += 15 unless page_view.new_record?
end
```

this way it only issues 2 queries, the `SELECT` and the `INSERT` and no `UPDATE` in any case.

Would that work?

Reply:
LOVE it!!!

Reply:
nvm this doesn't work for this scenario bc we will not increment the time_tracked_in_seconds if the record is found. It only increments in the event of a new record according to my testing.

Reply:
yeah, I can see that now :D well, let's keep it like it is!